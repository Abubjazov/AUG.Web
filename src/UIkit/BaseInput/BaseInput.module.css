.root {
  display: flex;
  flex-direction: column;
  gap: 3px;
  text-align: left;
  transition: all var(--transition-delay) linear;
  animation: var(--animation-visible);
}

.root input {
  padding: 10px;
  font-size: 16px;
  font-weight: 400;
  line-height: 16px;
  color: var(--logotext);
  border: 1px;
  border-radius: 10px;
  box-shadow: var(--box-shadow);
}

.error-message {
  font-size: 12px;
  font-weight: 300;
  line-height: 13px;
  color: var(--red);
  animation: var(--animation-visible);
}

.error-message-white {
  color: var(--white);
}
