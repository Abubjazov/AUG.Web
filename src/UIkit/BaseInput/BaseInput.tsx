import { FC } from 'react'

import { combineClasses as cc } from 'utils/combineClasses/combineClasses'

import styles from './BaseInput.module.css'

export interface BaseInputProps {
  dataTestId: string
  type: string
  name: string
  placeholder: string
  value: string | number | undefined
  onChange: React.ChangeEventHandler<HTMLInputElement> | undefined
  onBlur: React.FocusEventHandler<HTMLInputElement> | undefined
  errors: string[]
  errorWhite?: boolean
  isDirty: boolean
}

const BaseInput: FC<BaseInputProps> = ({
  dataTestId,
  type,
  name,
  placeholder,
  value,
  onChange,
  onBlur,
  errors,
  errorWhite,
  isDirty,
}) => {
  return (
    <div className={styles.root}>
      <input
        data-testid={dataTestId}
        type={type}
        name={name}
        placeholder={placeholder}
        value={value}
        onChange={onChange}
        onBlur={onBlur}
      />

      {isDirty && errors && (
        <span
          className={cc([
            styles['error-message'],
            errorWhite ? styles['error-message-white'] : '',
          ])}
        >
          {errors[errors.length - 1]}
        </span>
      )}
    </div>
  )
}

export default BaseInput
