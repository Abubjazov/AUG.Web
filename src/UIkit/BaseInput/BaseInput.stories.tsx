import type { Meta, StoryObj } from '@storybook/react'

import BaseInput, { BaseInputProps } from './BaseInput'

import '/src/index.css'

const meta: Meta<BaseInputProps> = {
  component: BaseInput,
  title: 'UIkit/BaseInput',
  parameters: {
    docs: {
      description: {
        component: 'Standard interface component - BaseInput',
      },
    },
  },
  tags: ['autodocs'],
  argTypes: {
    dataTestId: {
      description: 'HTML-input tag attribute: "data-testid"',
    },
    type: {
      description: 'HTML-input tag attribute: "type"',
    },
    name: {
      description: 'HTML-input tag attribute: "name"',
    },
    placeholder: {
      description: 'HTML-input tag attribute: "placeholder"',
    },
    value: {
      description: 'Input value',
    },
    onChange: {
      description: 'OnChange event handler function',
    },
    onBlur: {
      description: 'onBlur event handler function',
    },
    errors: {
      description:
        'An array containing all input errors currently occurring (the last element of the array is displayed)',
    },
    errorWhite: {
      description: 'Change the error display color to white',
    },
    isDirty: {
      description: 'Whether the element was touched by the user',
    },
  },
}

export default meta

type Story = StoryObj<BaseInputProps>

export const Default: Story = {
  args: {
    placeholder: 'placeholder text',
    onChange: undefined,
    onBlur: undefined,
    errors: ['Some Error'],
    isDirty: true,
  },
  decorators: [Story => <Story />],
}
