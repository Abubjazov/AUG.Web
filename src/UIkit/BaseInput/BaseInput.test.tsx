import { render } from '@testing-library/react'

import BaseInput from '.'

describe('BaseInput', () => {
  test('should render BaseInput default', () => {
    expect(
      render(
        <BaseInput
          dataTestId={'base-input'}
          type={'text'}
          name={'base-input-text'}
          placeholder={'Enter text'}
          value={undefined}
          onChange={undefined}
          onBlur={undefined}
          errors={[]}
          isDirty={false}
        />,
      ),
    ).toMatchSnapshot()
  })

  test('should render BaseInput then isDirty & on input error', () => {
    expect(
      render(
        <BaseInput
          dataTestId={'base-input'}
          type={'text'}
          name={'base-input-text'}
          placeholder={'Enter text'}
          value={undefined}
          onChange={undefined}
          onBlur={undefined}
          errors={['Input error']}
          isDirty={true}
        />,
      ),
    ).toMatchSnapshot()
  })

  test('should render BaseInput then isDirty & on input error with "errorWhite"', () => {
    expect(
      render(
        <BaseInput
          dataTestId={'base-input'}
          type={'text'}
          name={'base-input-text'}
          placeholder={'Enter text'}
          value={undefined}
          onChange={undefined}
          onBlur={undefined}
          errors={['Input error']}
          errorWhite
          isDirty={true}
        />,
      ),
    ).toMatchSnapshot()
  })
})
