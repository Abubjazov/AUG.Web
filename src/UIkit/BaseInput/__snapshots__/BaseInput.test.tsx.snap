// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseInput > should render BaseInput default 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_root_672eb8"
      >
        <input
          data-testid="base-input"
          name="base-input-text"
          placeholder="Enter text"
          type="text"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_root_672eb8"
    >
      <input
        data-testid="base-input"
        name="base-input-text"
        placeholder="Enter text"
        type="text"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BaseInput > should render BaseInput then isDirty & on input error 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_root_672eb8"
      >
        <input
          data-testid="base-input"
          name="base-input-text"
          placeholder="Enter text"
          type="text"
          value=""
        />
        <span
          class="_error-message_672eb8 "
        >
          Input error
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_root_672eb8"
    >
      <input
        data-testid="base-input"
        name="base-input-text"
        placeholder="Enter text"
        type="text"
        value=""
      />
      <span
        class="_error-message_672eb8 "
      >
        Input error
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BaseInput > should render BaseInput then isDirty & on input error with "errorWhite" 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_root_672eb8"
      >
        <input
          data-testid="base-input"
          name="base-input-text"
          placeholder="Enter text"
          type="text"
          value=""
        />
        <span
          class="_error-message_672eb8 _error-message-white_672eb8"
        >
          Input error
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_root_672eb8"
    >
      <input
        data-testid="base-input"
        name="base-input-text"
        placeholder="Enter text"
        type="text"
        value=""
      />
      <span
        class="_error-message_672eb8 _error-message-white_672eb8"
      >
        Input error
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
